### PR TITLE
feat: product tour in homepage

### PR DESCRIPTION
**Pull request by Golodhros**
### Summary of Changes
- Fixes and consolidates outline styles for better a11y
- Adds nav-bar icon button styles
- Fixes feedback icon on nav bar
- Adds the new tour component
- Plugs tour component in the homepage
- Adds configuration options for the tour
- Updates caniuse-lite
- Tweaks eslint rules

### Why
Related to https://github.com/amundsen-io/rfcs/blob/master/rfcs/045-amundsen-tour-and-feature-walkthrough.md

### Tests
Added tests
Tested manually

### Screenshots
Programmatic start Icons:
Amundsen_-_Data_Discovery_Portal

Tour step:
Amundsen_-_Data_Discovery_Portal

### Documentation
Updated docs on icons
Added docs about the tour

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: frontend/amundsen_application/static/js/features/NavBar/index.tsx [hunk 1]
```diff
@@ -1,146 +1,211 @@
...
-            </div>
+              <span className="title-3">{getLogoTitle()}</span>
+            </Link>
+          </div>
+          {renderSearchBar(location)}
+          <div id="nav-bar-right" className="ml-auto nav-bar-right">
+            {generateNavLinks(getNavLinks())}
+            {hasPageTour && <ProductTourButton onClick={handleTourClick} />}
+            {feedbackEnabled() && <Feedback />}
+            {loggedInUser && indexUsersEnabled() && (
```
Comment (kristenarmes) [inline frontend/amundsen_application/static/js/features/NavBar/index.tsx:167]: From the screenshot, it looks like the user dropdown circle icon is higher and closer to the feedback icon than the rest of the links/icons in the nav bar are in relation to each other - I see that is how it currently is too, but it seems a bit more noticeable now being next to the icons that are all aligned and equally spaced from the nav links. what do you think about lowering it and bringing it right by a few px?

**Review by kristenarmes — APPROVED** [review]
LGTM! Awesome feature! Just had a small comment/question on the nav bar alignment

File: frontend/amundsen_application/static/js/features/NavBar/index.tsx (hunk 1 above)
Comment (Golodhros) [reply frontend/amundsen_application/static/js/features/NavBar/index.tsx:167]: Good call!
I will review that spacing right now!

**Review by allisonsuarez — APPROVED** [review]
LGTM, thanks for fixing the badges styling!

**Pull request merged**